Assets/Scripts/Models/Need.cs

```diff
@@ -83,25 +89,57 @@ public string DisplayAmount
         }
     }
 
+    /// <summary>
+    /// Gets the EventAction for the current furniture.
+    /// These actions are called when an event is called. They get passed the furniture
+    /// they belong to, plus a deltaTime (which defaults to 0).
+    /// </summary>
+    /// <value>The event actions that is called on update.</value>
+    public EventActions EventActions { get; private set; }
+
     // Update is called once per frame
     public void Update(float deltaTime)
     {
-        FunctionsManager.Need.CallWithInstance(luaUpdate, this, deltaTime);
-        if (luaOnly)
+        if (EventActions != null && EventActions.HasEvent("OnUpdate"))
         {
-            return;
+            EventActions.Trigger("OnUpdate", this, deltaTime);
         }
-
-        Amount += growthRate * deltaTime;
-
-        if (character != null && character.CurrTile.GetGasPressure("O2") < 0.15)
+        else 
         {
-            Amount += (addedInVacuum - (addedInVacuum * (character.CurrTile.GetGasPressure("O2") * 5))) * deltaTime;
+            DefaultNeedDecay(deltaTime);
         }
 
         if (Amount.AreEqual(100))
         {
-            // FIXME: Insert need fail damage code here.
+            if (EventActions != null && EventActions.HasEvent("OnEmptyNeed"))
+            {
+                EventActions.Trigger("OnEmptyNeed", this, deltaTime);
+            }
+            else
+            {
+                DefaultEmptyNeed();
+            }
+        } 
+        else if (Amount > 90f)
+        {
+            if (EventActions != null && EventActions.HasEvent("OnSevereNeed"))
```

Comment:
You're probably right, for now as far as checking if the event exists beforehand, though in the future I'd like to come up with default things, but I don't have a clear idea what all of the needs are going to want to do, and some of the what to do on needs is still tied up in Character, but I don't know if there's a good way with the current job system to have jobs to do for a specific character but that don't have to be done immediately ( such as you might want with moderate need ).

But for now I could probably remove the HasEvent Checks... though I wonder what people's opinions are on having Trigger() return a bool depending on if it performed any actions or not (so rather than checking first if one exists and then triggering if it does, and doing something else if it doesn't, we could just trigger and if it doesn't do anything we could then do something else.)